### ISSUE TITLE
Support specifying steps multiple times on the command line

### ISSUE THREAD

**Issue by pvalena** [teemtee/tmt#68]
If I run:
```
$ tmt --root examples/mini run --debug --all provision --how=local provision
```
the provision is actually run with default `how`, and `--how=local` is discarded.

I think all steps do behave this way; and I find it quite counter-intuitive.

Maybe a warning could be thrown at least?

_ _ _ _ 

_Note:_ if I run
```tmt --root examples/mini run --all provision --how=local execute discover finish```
it behaves as expected.
Adding the `provision` at the end results in the unexpected behaviour.

**Comment by psss** [comment]
The question is whether we want to support single step multiple times on the command line. I can imagine this could be useful. For example discover tests from multiple repositories:

    tmt run --all discover --url https://one.git.repo/ --revision master discover --url https://second.git.repo/ --revision devel

So shall we support this for all steps? Or selected only?

**Comment by pvalena** [comment]
I think we might want to support multiple ins, based on the step, but what would be the behaviour?

In case of `provision` it's simple, we have (planned) support for multiple provisions (`provision_dir`), same with `prepare`- but I'm not sure what of the others?

`execute` - do we have run multiple test suites? could it be like a second suite /plan/ ... just another folder (`two` instead of `one`; in the default naming scheme) and we simply run another `run.sh` .... on the second folder? Then we'd have to aggregate the results though.

_ _ _ _

In general, what's preventing us to have it for all?

**Comment by psss** [comment]
For the `discover` step it makes sense: You could have mutliple repositories with tests or some tests stored directly with the app code, some shared in a different repository. For the `report` step you might want to send the report to multiple destinations. So it seems to functionality could be useful for more steps. Make sense to implement this in a similar / consistent way for all steps.

**Comment by psss** [comment]
In addition to providing the step config multiple times on the command line we would also like to be able to do the following three actions:

 - completely override plan config (should be probably default)
 - insert/append a new step config (by providing `order`)
 - update an existing config (match by `name`)

**Comment by lukaszachy** [comment]
In the end we probably need to support four actions: 
* completely override plan config (should be probably default)
* insert/append a new step config (by providing order)
* update an existing config (match by name, local defaults have lower prio)
* weak update (only set values which are not set, defaults have higher prio)

**Comment by sopos** [comment]
Currently, the commad line options wins completely. To change it to and an update or an append operation rather than a replace operation the syntax could be extended with an fmf-like style `tmt run provision+ ...` to express an addition to the provision in the same way as fmf inheritance works.

**Comment by thrix** [comment]
I would propose to use

* `run -a provision --image centos-7` would be weak update by default
* `run -a provision --image centos-7 --force-image` would be forced for all

**Comment by psss** [comment]
This is one of the important blockers for the multihost testing support. It seems that `click` should support providing multiple subcommands so the implementation should not be that hard. Let's try to implement this in `1.22`. Any volunteer? ;-)

**Comment by thrix** [comment]
@happz please check if this is being resolved with your patches

**Comment by happz** [comment]
Partially, this has been resolved: in 1.27 we should get `--update` and `--insert` for phases on the command line, allowing us to specify multiple phases and overwrite their keys. `--weak-update` has not been implemented in the 1.27 window.

I filed https://github.com/teemtee/tmt/issues/2285 to track the `--weak-update`, and I'd propose closing this issue in 1.27. Did I forget anything else?

**Comment by thrix** [comment]
Closing as resolved via #2188, the issue #2285 will track leftovers.
